### PR TITLE
feat(datasets): spans as examples

### LINKED ISSUE #3278: [datasets][gql] span as example

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #3278 

adds the ability to view a span as an example before committing it to a dataset

```graphql
query projectSpans {
  node(id: "UHJvamVjdDox") {
    ... on Project {
      spans {
        edges {
          span: node {
            ... on Span {
              asExample {
                input
                output
                metadata
              }
            }
          }
        }
      }
    }
  }
}
```

### CODE REVIEW

File: src/phoenix/server/api/helpers/dataset_helpers.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
+from typing import Any, Dict, Literal, Optional, Protocol
```
Comment (mikeldking) [inline src/phoenix/server/api/helpers/dataset_helpers.py:1]: just moved to helpers directory

File: src/phoenix/server/api/types/Span.py [hunk 2]
```diff
@@ -303,3 +343,21 @@ def _convert_metadata_to_string(metadata: Any) -> Optional[str]:
         return json.dumps(metadata)
     except Exception:
         return str(metadata)
+
+
+@dataclass
```
Comment (axiomofjoy) [inline src/phoenix/server/api/types/Span.py:348]:
```suggestion
@dataclass(frozen=True)
```

Suggestion: Immutable is a safer default.

File: src/phoenix/server/api/types/Span.py [hunk 3]
```diff
@@ -189,6 +203,31 @@ async def descendants(
...
+    @strawberry.field(
+        description="The span's attributes translated into an example for a dataset",
+    )  # type: ignore
+    def as_example(self) -> SpanAsExample:
+        db_span = self.db_span
+        attributes = db_span.attributes
+        span_io = _SpanIO(
+            span_kind=db_span.span_kind,
+            input_value=get_attribute_value(attributes, INPUT_VALUE),
+            input_mime_type=get_attribute_value(attributes, OUTPUT_MIME_TYPE),
```
Comment (axiomofjoy) [inline src/phoenix/server/api/types/Span.py:215]:
```suggestion
            input_mime_type=get_attribute_value(attributes, INPUT_MIME_TYPE),
```

File: src/phoenix/server/api/types/Span.py [hunk 4]
```diff
@@ -101,9 +110,14 @@ def from_dict(
         )
 
 
+@strawberry.type
+class SpanAsExample(Example): ...
```
Comment (axiomofjoy) [inline src/phoenix/server/api/types/Span.py:114]: Do we want this in a separate file with the other types?

**Review by axiomofjoy — CHANGES_REQUESTED** [review]
Need to swap out `INPUT_MIME_TYPE`.

File: src/phoenix/server/api/types/Span.py (hunk 4 above)
Comment (mikeldking) [reply src/phoenix/server/api/types/Span.py:114]: This is only relevant to this file so I think it's okay. I can move the interface out if needed.

File: src/phoenix/server/api/helpers/dataset_helpers.py
[the remark this answers is not in the document]
Comment (mikeldking) [reply src/phoenix/server/api/helpers/dataset_helpers.py:1]: I thought so. Not sure why my tests passed ❓

File: src/phoenix/server/api/types/DatasetExample.py
[the remark this answers is not in the document]
Comment (mikeldking) [reply src/phoenix/server/api/types/DatasetExample.py:1]: I think SpanAsExample is too small of a type to warrant a file for now

File: src/phoenix/server/api/helpers/dataset_helpers.py
[the remark this answers is not in the document]
Comment (mikeldking) [reply src/phoenix/server/api/helpers/dataset_helpers.py:1]: Done

File: src/phoenix/server/api/types/Span.py (hunk 3 above)
Comment (mikeldking) [reply src/phoenix/server/api/types/Span.py]: 🤦🏻 thanks

File: src/phoenix/server/api/types/Span.py (hunk 2 above)
Comment (mikeldking) [reply src/phoenix/server/api/types/Span.py]: 👍🏻

Comment (mikeldking) [reply src/phoenix/server/api/types/Span.py:348]: Actually I can't do this as it doesn't comply with the HasSpanIO protocol

File: tests/server/api/mutations/test_dataset_mutations.py
Comment (axiomofjoy) [inline tests/server/api/mutations/test_dataset_mutations.py:1]: Can we move the `test_get_dataset_example_output` and `test_get_dataset_example_input` test cases into a `test_dataset_helpers.py` file to reflect the new location of the corresponding functions?

**Review by axiomofjoy — APPROVED** [review]

**Pull request merged**